arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx

```diff
@@ -162,6 +163,109 @@ You should see something like this:
+
+```go
+// ArbSys provides system-level functionality for interacting with L1 and understanding the call stack.
+type ArbSys struct {
+	Address                 addr // 0x64
+	L2ToL1Tx                func(ctx, mech, addr, addr, huge, huge, huge, huge, huge, huge, []byte) error
+	L2ToL1TxGasCost         func(addr, addr, huge, huge, huge, huge, huge, huge, []byte) (uint64, error)
+	SendMerkleUpdate        func(ctx, mech, huge, bytes32, huge) error
+	SendMerkleUpdateGasCost func(huge, bytes32, huge) (uint64, error)
+	InvalidBlockNumberError func(huge, huge) error
+
+	// deprecated event
+	L2ToL1Transaction        func(ctx, mech, addr, addr, huge, huge, huge, huge, huge, huge, huge, []byte) error
+	L2ToL1TransactionGasCost func(addr, addr, huge, huge, huge, huge, huge, huge, huge, []byte) (uint64, error)
+
+	// Add your customize event here:
+	Hi						func(ctx, mech, addr) error
+	// This is needed which will tell you how much gas it will cost, the param is the same as your event but without the first two (ctx, mech), the return param is always (uint64, error)
+	HiGasCost				func(addr) (uint64, error)
+}
+```
+
+Then add the event to `SayHi` method:
+
+```go
+func (con *ArbSys) SayHi(c ctx, evm mech) (string, error) {
+	err := con.Hi(c, evm, c.caller)
+	return "hi", err
+}
+```
+
+Now navigate to the [precompiles interface][precompile_interface_dir_link] directory, open `Arbsys.sol`, and add the required interface. Ensure that the Event name on the interface matches the name of the function you introduced in `ArbSys` struct in the previous step:
+
+```solidity
+event Hi(address caller);
+```
+
+If you want to make the param in event as `index`, just add index to solidity interface as this:
+
+```solidity
+event Hi(address indexed caller);
+```
+
+Now as emiting events need to cost gas, so we should remove `view` function behavior from the method:
+
+```solidity
+function sayHi() external returns(string memory);
+```
+
+Next, build Nitro by following the instructions in [How to build Nitro locally](/node-running/how-tos/build-nitro-locally). Note that if you've already built the Docker image, you still need run the last step to rebuild.
+
+Run Nitro with the following command:
+
+```shell
+docker run --rm -it  -v /some/local/dir/arbitrum:/home/user/.arbitrum -p 0.0.0.0:8547:8547 -p 0.0.0.0:8548:8548 @latestNitroNodeImage@ --parent-chain.connection.url=<YourParentChainUrl> --chain.id=<YourOrbitChainId> --http.api=net,web3,eth,debug --http.corsdomain=* --http.addr=0.0.0.0 --http.vhosts=*
+```
+
+:::info
+
+Note that the instructions provided in [How to run a full node](/node-running/how-tos/running-a-full-node) **will not** work with your Orbit node. See [Command-line options (Orbit)](/launch-orbit-chain/reference/command-line-options) for Orbit-specific CLI flags.
+
+:::
+
+### Send the transaction and get the transaction receipt
+
+Send transaction to `ArbSys`, note as the function is not view/pure function, we need to send transaction with gas cost:
```

Comment:
```suggestion
Send a transaction to `ArbSys`, note that since this isn't a view or pure function, you'll need to send a transaction with associated gas costs:
```